style: put a paintbrush on the site

Favna opened the pull request:
It looks so much better now and the UX is much better too.

~~also it looks more discord-y~~

<details><summary> Screenshots inside galore (click screenshot for full size version)! </summary>


<h1> Home Page </h1>

<a target="_blank" href="https://user-images.githubusercontent.com/4019718/72226334-4ce72180-3590-11ea-8412-5e16915a1b1e.png"><img src="https://user-images.githubusercontent.com/4019718/72226334-4ce72180-3590-11ea-8412-5e16915a1b1e.png" width=500 /></a>

<h1> Words filter page </h1>

<a target="_blank" href="https://user-images.githubusercontent.com/4019718/72226348-7607b200-3590-11ea-8f1f-5742e45b5cb0.png"><img src="https://user-images.githubusercontent.com/4019718/72226348-7607b200-3590-11ea-8f1f-5742e45b5cb0.png" width=500 /></a>



</details>

gc commented:
I dont think you should add 2 new libraries and rewrite half the sites styling just to make a couple tiny visual changes. Secondly I'm not a big fan of copying discords exact style/UI and calling it an improvement, it looks tacky and generic, 90% of 3rd party discord-related website do this. I'm against this being merged for those 2 reasons.

Favna commented:
@GC as thoroughly clearly stated in Discord, but you seem to be completely missing that point, this does NOT only do styling. It also has various bug fixes, some very important and totally breaking ones. You need just start master locally once, pop open the console and you're directly presented with errors.

As for this whole "two new dependencies" stuff, we went over React IF thoroughly just last week. Don't tell me that was your alternate personality that approved of the library. And as for node-sass, it's a ***dev***depdency for christs sake. There's being nitpicky and there's being paranoid.

But if you're going to be so adamant about not merging this PR then please by all means do fix all the bugs I fixed in this PR in another one but so help me god if I find one bug I fixed here that is NOT fixed in that PR man.

Favna commented:
Full changelog as taken from the commit message body:

- style: fix styling on homepage
- fix: close sidebar on click on mobile
- refactor: layout fixes & cleanup stuff
- meta: replace all instances of kyranet/skyra.pw with skyra-project/skyra.pw
- ci: use v2 of actions/checkout
---
- Fixed issue that made the slider thumbnail inconsistent across screen sizes
- Fixed issue that made the duration/unit FormControl's inconsistent across screen sizes
- Moved src/pages/Dashboard/components to src/components
- Moved all "Select" related components from src/components to src/components/Select/
- Rewrote SelectDuration to TSX to provide better intellisense
- Rewrote Select to TSX to provider better intellisense
- Organized imports so they are consistent across all files
- Disabled "function-paren-newline" ESLint rule as it is covered by Prettier
- Added "xs" media breakpoint to _breakpoints.scss
- And renamed the variables used therein to better represent their usage
- Renamed src/pages/Dashboard/Filter/Index.js to Words.js to better represent what the file is and to avoid it being the index of the folder when importing
